tap_hubspot_beta/tap.py

```diff
@@ -166,7 +168,15 @@ def __init__(
 
     def discover_streams(self) -> List[Stream]:
         """Return a list of discovered streams."""
-        return [stream_class(tap=self) for stream_class in STREAM_TYPES]
+        streams = [stream_class(tap=self) for stream_class in STREAM_TYPES]
+        try:
+            discover_stream = DiscoverCustomObjectsStream(tap=self)
+            for record in discover_stream.get_records(context={}):
+                stream_class = self.generate_stream_class(record)
+                streams.append(stream_class(tap=self))
+        except FatalAPIError as exc:
+            self.logger.info(f"failed to discover custom objects. Error={exc}")
```

Comment:
Should the try-except block be inside the for loop, as each record can generate a new `stream_class`? Can we get duplicate records here? I would log it as a warning and not as info.

Reply:
None of the errors you are raising in `generate_stream_class` will be caught on this except that you included. Is that intended?

Reply:
the line that raise this error is 174. it is related with permissions to use custom objects, not to discover/generate schema